OpenRA.Game/Traits/Player/Shroud.cs

```diff
@@ -200,6 +199,19 @@ void ITick.Tick(Actor self)
 					var uv = touched.PPosFromIndex(index);
 					if (map.Contains(uv))
 						OnShroudChanged(uv);
+
+					if (ExploreMapEnabled)
+						RevealedCells = map.ProjectedCells.Length;
```

Comment:
As i recall when a spectator/observer changes shroud settings a new shroud will be created - (with different visibility functions).

Please don't update/calculate this stat each tick. This method is a performance bottleneck.